src/screenbot.js

```diff
@@ -49,22 +91,31 @@ var Screenbot = (function Screenbot() {
         _this.command = function (command, cb) {
           console.log("Source: " + _private.source);
 
-          _request(_endpoint(command), function(success, response) {
-
-            if (!success) return cb(false, response.error);
+          _request(_endpoint(command), function(err, result) {
+            if(err instanceof Error) return cb(err);
+            if(err) return cb(_handleError({ code: ERROR_CODES.CLIENT_UNAVAILABLE.code, message: err.error }));
+            if('connected' in result) return cb(null, { connected: result.connected });
 
             if(_private.source && _private.source.readyState !== 2) {
               _private.source.close();
             }
-            _private.source = new EventSource(_response_endpoint(response.token));
+
+            _private.source = new EventSource(_response_endpoint(result.token));
             _private.source.onmessage = function(event) {
               _private.source.close();
-              if(event.data === "0") cb(false);
-              if(event.data !== "0") cb(true, event.data);
+              if(event.data === "0" || !event.data.length) return cb(_handleError({ code: ERROR_CODES.NO_DATA_RECEIVED.code }));
+
+              var eventData = { url: event.data };
+              cb(null, eventData);
             };
           });
         };
 
+        // Check whether the Screenbot app is running and connected
+        _this.ping = function(cb) {
+          _this.command("ping",cb);
+        };
```

Comment:
This could be done without repeating code.

```javascript
// Up above somewhere...

// These commands available as methods
var COMMANDS = [
  'ping', // Check whether the Screenbot app is running and connected
  [ activate, '' ], // Take a screenshot (aliased)
  'shot', // Take a screenshot
  'draw', // Create an annotation
  'cast', // Create a screencast
  'clip' // Upload your clipboard
];

function generateCurriedCommand(command) {
  return function(cb) { _this.command(command, cb) };
}

// In this space...

// Export each command's method
COMMANDS.forEach(function(cmd) {
  if(cmd instanceof Array)
    _this[cmd[0]] = generateCurriedCommand(cmd[1]);
  else
    _this[cmd] = generateCurriedCommand[cmd]);
});
```

It's not really shorter but it provides a more explicit definition of these curried methods.